overlord/auth/auth.go

```diff
@@ -69,14 +122,29 @@ func NewUser(st *state.State, username, email, macaroon string, discharges []str
 		return nil, err
 	}
 
+	if authStateData.MacaroonKey == nil {
+		authStateData.MacaroonKey, err = generateMacaroonKey()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	localMacaroon := macaroon
+	if username != "" {
```

Comment:
Right. Note what it conditional here is the macaroon setup for local user (if no username is provided, the store one is used). In any case, switching macaroons to use the user.ID, it shouldn't be a problem if we have a username or not to setup a snapd macaroon then. Updating this too.